readme.md
# Gravity Fieldset for Gravity Forms

## Description

Forked from [Gravity Fieldset for Gravity Forms](https://wordpress.org/plugins/gravity-fieldset-for-gravity-forms/) to fix issue with removing fieldset end, does not seem to be active any more.

> This plugin is an add-on for the [Gravity Forms](http://www.gravityforms.com).

Features of this plugin:

* Adding real fieldsets around your form fields
* Includes option for fieldset legend
* Autocomplete of fieldsets to properly close all HTML tags
* Auto-delete of fieldsets to properly close all HTML tags

Available Languages:

* English
* Dutch

## Installation

1. This plugin requires the Gravity Forms plugin, installed and activated
2. Upload folder to the `/wp-content/plugins/` directory
3. Activate the plugin through the 'Plugins' menu in the WordPress administration
4. Your fieldsets are now available in your form editor

CHANGELOG
== Changelog ==

= 0.3 =
* Forked 

= 0.2.2 =
* Fix: Delete closing fieldset

= 0.2.1 =
* Fix: Text Domain

= 0.2 =
* Added support for custom classes in fieldset end
* Major improvement in HTML output
* Fix: Compatability issue with Gravity PDF plugin
* Fix: compatability issue with WCAG 2.0 form fields for Gravity Forms plugin
* Fix: Compatability issue with Gravity Forms Repeater Add-On plugin

PLEASE NOTE:
class .gform_fieldset_end has been removed
class .gform_fieldset_begin will be deprecated in the next release

= 0.1 =
* Cleanup
* Added comments
* Improved English and Dutch Language
* Added auto close or open when adding fieldset begin or end
* Added auto delete of fieldset end when deleting fieldset begin

= 0.0.1 =
* First release

js/raak_fieldset_admin.js
jQuery(document).ready(function($) {
	
	fieldSettings["FieldsetBegin"] = ".label_setting, .css_class_setting, .conditional_logic_field_setting";
	fieldSettings["FieldsetEnd"] = ".css_class_setting";
	
	function fieldsetExist() {
		
		var fieldsetsCount = jQuery( '#gform_fields .gform_fieldset' ).length;
		
		return fieldsetsCount;
		
	}
	
	jQuery(document).bind('gform_field_deleted', function( event, form, field ){
	
		var fieldsetClosed = true;
		
		jQuery.each( form.fields, function( index, value ) {
			
			if ( typeof value.type != "undefined" ) {
				
				if ( value.type == 'FieldsetBegin' ) {
					
					fieldsetClosed = false;
					
				} else if ( value.type == 'FieldsetEnd' ) {
					
					console.log( value );
					
					if ( fieldsetClosed ) {
						
						deleteFieldset( value.id );
						
						return;
						
					}
					
					fieldsetClosed = true;
					
				}
				
			}
			
		});
	
	});
	
	jQuery(document).bind( 'gform_field_added', function( event, form, field ) {
		
		if ( field['type'] == 'FieldsetBegin' || field['type'] == 'FieldsetEnd' ) {
			
			var fieldsetClosed = true;
			var index = 1;
			
			jQuery.each( form.fields, function( index, value ) {
				
				if ( typeof value.type != "undefined" ) {
					
					if ( value.type == 'FieldsetBegin' ) {
						
						if ( fieldsetClosed ) {
							
							fieldsetClosed = false;
							
						} else {
							
							StartAddField( 'FieldsetEnd', index );
							
							fieldsetClosed = true;
							
							return;
							
						}
						
					} else if ( value.type == 'FieldsetEnd' ) {
						
						if ( fieldsetClosed ) {
						
							StartAddField( 'FieldsetBegin', index );
							
							return;
							
						} else {
							
							fieldsetClosed = true;
							
						}
						
					}
					
				}
				
				index++;
				
			});
			
			if ( !fieldsetClosed ) {
				
				StartAddField( 'FieldsetEnd' );
				
			}
			
		}
	
	} );
	
});

gravity-fieldset-for-gravity-forms-fork.php
<?php
/*
Plugin Name: Gravity Fieldset for Gravity Forms Fork
Version: 0.3
Description: Extends the Gravity Forms plugin - adding a fieldset open and close field that can be used to create 'real' sections.
Author: Bas van den Wijngaard & Harro Heijboer forked by Ben Freeman
Text Domain: gravity-fieldset-for-gravity-forms
Domain Path: /languages
License: GPL2 v2
*/



/**
 * Load translations
 */

function raak_gf_fieldset_load_textdomain() {
	
	load_plugin_textdomain( 'gravity-fieldset-for-gravity-forms', FALSE, basename( dirname( __FILE__ ) ) . '/languages' );
	
}

add_action( 'plugins_loaded', 'raak_gf_fieldset_load_textdomain', 1 );

add_action( 'admin_notices', array('RAAK_GF_Fieldset', 'admin_warnings' ), 20 );



/**
 * RAAK_GF_Fieldset class.
 */

if (!class_exists('RAAK_GF_Fieldset')) :

	
	class RAAK_GF_Fieldset
	{
		
		private static $name = 'Fieldset for Gravity Forms';
		private static $slug = 'raak_gf_fieldset';
		private static $version = '0.2';
		
		
		/**
		 * Construct the plugin object
		 */
		
		public function __construct()
		{
			
			// register plugin functions through 'plugins_loaded' -
			// this delays the registration until all plugins have been loaded, ensuring it does not run before Gravity Forms is available.
			
			add_action( 'plugins_loaded', array( &$this, 'register_actions' ) );
			
		}
		
		
		/*
		 * Register plugin functions
		 */
		
		function register_actions()
		{
			
			// register actions
			
			if (self::is_gravityforms_installed()) :
				
				// start plug in
				
				// add buttons to the GF
				
				add_filter( 'gform_add_field_buttons', array( &$this, 'fieldset_add_field' ) );
				
				// add input field for field title
				
				add_filter( 'gform_field_type_title' , array( &$this, 'fieldset_title' ), 10, 2 );
				add_action( 'gform_editor_js', array( &$this, 'fieldset_custom_scripts' ) );
				add_action( 'gform_field_css_class', array( &$this, 'fieldset_custom_class' ), 10, 3 );
				add_filter( 'gform_field_content', array( &$this, 'fieldset_display_field' ), 10, 5 );
				
				
				// add filter for altering the fieldset container html
				add_filter( "gform_field_container", array( &$this, 'filter_gform_field_container'), 10, 6 );
				add_filter( 'gform_field_content', array( &$this, 'filter_gform_field_remove_label'), 10, 6 );
				
				// add filter for altering the complete form HTML
				add_filter( 'gform_get_form_filter', array( &$this, 'filter_gform_cleanup_html' ), 10, 2 );
				
			endif;
			
		}
		
		
		/**
		 * Create a new fields group in the Gravity Forms forms editor and add our fieldset 'fields' to it.
		 */
		
		public static function fieldset_add_field( $field_groups )
		{
			
			// add begin fieldset button
			
			$fieldset_begin_field = array(
				
				'class'		=> 'button',
				'value'		=> __( 'Fieldset Begin', 'gravity-fieldset-for-gravity-forms' ),
				'data-type'	=> 'FieldsetBegin',
				'onclick'	=> 'StartAddField( \'FieldsetBegin\' );'
				
			);
			
			// add end fieldset button
			
			$fieldset_end_field = array(
				
				'class'		=> 'button',
				'value'		=> __( 'Fieldset End', 'gravity-fieldset-for-gravity-forms' ),
				'data-type'	=> 'FieldsetEnd',
				'onclick'	=> 'StartAddField( \'FieldsetEnd\' );'
				
			);

			foreach ( $field_groups as &$group ) :
				
				$raak_fields_active = false;

				if ( $group["name"] === "raak_fields" ) :
					
					$raak_fields_active = true;
					
					$group["fields"][] = $fieldset_begin_field;
					$group["fields"][] = $fieldset_end_field;
					
				endif;

			endforeach;

			if ( !$raak_fields_active ) :
				
				$field_groups[] = array(
					
					'name'		=> 'raak_fields',
					'label'		=> __( 'Fieldsets', 'gravity-fieldset-for-gravity-forms' ),
					'fields'	=> array( $fieldset_begin_field, $fieldset_end_field )
					
				);
				
			endif;

			return $field_groups;
			
		}
		
		
		/**
		 * Add title to fieldset, displayed in Gravity Forms forms editor
		 */
		
		public static function fieldset_title( $title, $field_type )
		{
			
			if ( $field_type === "FieldsetBegin" ) :
				
				return __( 'Fieldset Begin', 'gravity-fieldset-for-gravity-forms' );
				
			elseif ( $field_type === "FieldsetEnd" ) :
			
				return __( 'Fieldset End', 'gravity-fieldset-for-gravity-forms' );
				
			else :
			
				return __( 'Unknown', 'gravity-fieldset-for-gravity-forms' );
				
			endif;
			
		}
		
		
		/**
		 * JavaSript to add field options to fieldset fields in the Gravity forms editor
		 */
		
		public static function fieldset_custom_scripts()
		{
			
			// add custom css
			
			wp_enqueue_style(
				
				'raak_fieldset_admin_style',
				plugins_url( '/css/raak_fieldset_admin.css', __FILE__ )
				
			);
			
			// add js
			?>
			
			<script type="text/javascript">
			
			<?php
			// include JS that do not require PHP parse
			
			include( plugin_dir_path( __FILE__ ) . '/js/raak_fieldset_admin.js' );
			
			// include JS that requires PHP parsing
			
			include( plugin_dir_path( __FILE__ ) . '/js/raak_fieldset_admin.php' );
			?>
			
			</script>
			
			<?php
			
		}
		
		
		/**
		 * Add custom classes to fieldset fields, controls CSS applied to field
		 */
		
		public static function fieldset_custom_class($classes, $field, $form)
		{
			
			if ( $field['type'] === 'FieldsetBegin' ) :
				
				$classes .= ' gform_fieldset_begin gform_fieldset';
				
			elseif ($field['type'] === 'FieldsetEnd') :
			
				$classes .= ' gform_fieldset_end gform_fieldset';
				
			endif;

			return $classes;
			
		}
		
		
		/**
		 * Displays fieldset
		 */
		
		public static function fieldset_display_field( $content, $field, $value, $lead_id, $form_id )
		{
			
			$custom_field_classes = $field->cssClass;
			
			if ( ( !is_admin() ) && ( $field['type'] == 'FieldsetBegin') ) :
				
				$content .= '<fieldset class="gfieldset gform_fieldset_begin gform_fieldset '.$custom_field_classes.'">';

				if ( isset( $field['label'] ) && trim( $field['label'] ) !== '' ) :
					
					$content .= '<legend class="gfieldset-legend">' . trim( $field['label'] ) . '</legend>';
					
				endif;

			elseif ( ( !is_admin() ) && ( $field['type'] == 'FieldsetEnd' ) ) :
				
				$content .= '</fieldset>';
				
			endif;

			return $content;
			
		}
		
		
		/*
		 * Warning message if Gravity Forms is installed and enabled
		 */
		
		public static function admin_warnings()
		{
			
			if ( !self::is_gravityforms_installed() ) :
				
				$message = __('requires Gravity Forms to be installed.', 'gravity-fieldset-for-gravity-forms');
				
			endif;

			if ( empty( $message ) ) return;
			?>
			
			<div class="error">
				
				<h3>Warning</h3>
				
				<p><?php _e('The plugin', 'gravity-fieldset-for-gravity-forms'); ?> <strong><?php echo self::$name; ?></strong> <?php echo $message; ?><br /><?php _e('Please', 'gravity-fieldset-for-gravity-forms'); ?> <a target="_blank" href="http://www.gravityforms.com/"><?php _e('download the latest version', 'gravity-fieldset-for-gravity-forms'); ?></a> <?php _e('of Gravity Forms and try again.', 'gravity-fieldset-for-gravity-forms'); ?></p>
				
			</div>
			
			<?php
		}
		
		
		/*
		 * Check if GF is installed
		 */
		
		private static function is_gravityforms_installed()
		{
			if ( !function_exists( 'is_plugin_active' ) || !function_exists( 'is_plugin_active_for_network' ) ) :
				
				require_once( ABSPATH . '/wp-admin/includes/plugin.php' );
				
			endif;

			if (is_multisite()) :
				
				return (
					is_plugin_active_for_network( 'gravityforms/gravityforms.php' ) ||
					is_plugin_active( 'gravityforms/gravityforms.php' )
				);
				
			else :
				
				return is_plugin_active( 'gravityforms/gravityforms.php' );
				
			endif;
			
		}
		
		
		/*
		 * Alter container html when field type is fieldset
		 */
		
		public static function filter_gform_field_container( $field_container, $field, $form, $css_class, $style, $field_content )
		{

			$ul_classes = GFCommon::get_ul_classes($form);
			
			if ( ( !is_admin() ) && ( $field->type === 'FieldsetBegin' || $field->type === 'FieldsetEnd' ) ) :

				$field_container = '</ul>{FIELD_CONTENT}<ul class="'.$ul_classes.'">';
			
			endif;
			
			return $field_container;
		}

		
		/*
		 * Remove label tag when field type is fieldset
		 */
		
		public static function filter_gform_field_remove_label( $field_content, $field, $value, $lead_id, $form_id )
		{
			
			if ( ( !is_admin() ) && ( $field->type === 'FieldsetBegin' || $field->type === 'FieldsetEnd' ) ) :
			
				$field_content = preg_replace( '/<label[^>]*>([\s\S]*?)<\/label[^>]*>/', '', $field_content );
			
			endif;
			
			return $field_content;
			
		}
		
		
		/*
		 * Remove empty ul tag that is created when the fieldset close type is the last formfield.
		 */
		
		public static function filter_gform_cleanup_html( $form_string, $form )
		{
			
			if ( !is_admin() ) :
				
				$form_string = preg_replace( '#<(ul+)[^>]*>([[:space:]]|&nbsp;)*</ul>#', '', $form_string );
						
			endif;
			
			return $form_string;
			
		}
		
	}

	$RAAK_GF_Fieldset = new RAAK_GF_Fieldset();

endif;
